cmd/plugin_test.go

```diff
@@ -66,9 +188,32 @@ func TestStashPluginMain(t *testing.T) {
 	defer os.Remove(tempDir)
 
 	// Parts of test adapted from: https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go
+	// Basically, we need to run the test like this since StashPluginMain calls os.Exit() which is not good for our tests
+	// and leaves xrootd running. To work with this, we wrap the test in its own command and parse the output for successful run
 	if os.Getenv("RUN_STASHPLUGIN") == "1" {
+		viper.Set("Origin.EnablePublicReads", true)
+		fed := FedTest{T: t}
+		fed.Spinup()
+		defer fed.Teardown()
+
+		testFileContent := "test file content"
+		// Drop the testFileContent into the origin directory
+		tempFile, err := os.Create(filepath.Join(fed.OriginDir, "test.txt"))
+		assert.NoError(t, err, "Error creating temp file")
+		defer os.Remove(tempFile.Name())
```

Comment:
`Teardown` already removes all files under `f.OriginDir` so no need to do it separately